public/components/SavedQueriesPopover/SavedQueriesPopover.tsx

```diff
@@ -0,0 +1,196 @@
+  const [selectedQuery, setSelectedQuery] = useState<SavedQuery | null>(null);
+
+  // Fetch saved queries when popover opens
+  useEffect(() => {
+    if (isOpen && savedQueryService) {
+      setIsLoading(true);
+      savedQueryService
+        .getAllSavedQueries()
+        .then((queries: SavedQuery[]) => {
+          // Filter to only PPL queries
+          const pplQueries = queries.filter(
+            (q) => q.attributes?.query?.language?.toLowerCase() === 'ppl'
+          );
+          setSavedQueries(pplQueries);
+          
+          // Convert to selectable options
+          const opts: EuiSelectableOption[] = pplQueries.map((q) => ({
+            label: q.attributes.title || 'Untitled',
+            key: q.id,
+            prepend: '📝',
+            append: q.attributes.description ? (
+              <EuiText size="xs" color="subdued">
+                {q.attributes.description}
+              </EuiText>
+            ) : undefined,
+          }));
+          setOptions(opts);
+          setIsLoading(false);
+        })
+        .catch((err: any) => {
+          console.error('[SavedQueriesPopover] Error fetching saved queries:', err);
+          setIsLoading(false);
+        });
+    }
+  }, [isOpen, savedQueryService]);
+
+  const handleSelectionChange = useCallback(
+    (newOptions: EuiSelectableOption[]) => {
+      const selected = newOptions.find((opt) => opt.checked === 'on');
+      if (selected) {
+        const query = savedQueries.find((q) => q.id === selected.key);
+        setSelectedQuery(query || null);
+      } else {
+        setSelectedQuery(null);
+      }
+      setOptions(newOptions);
+    },
+    [savedQueries]
+  );
+
+  const handleLoadClick = useCallback(() => {
+    if (selectedQuery) {
+      const queryText = selectedQuery.attributes?.query?.query || '';
+      onLoadQuery(queryText);
+      onClose();
+    }
+  }, [selectedQuery, onLoadQuery, onClose]);
+
+  const button = (
+    <EuiButtonEmpty
+      size="s"
+      iconType={isOpen ? 'arrowUp' : 'arrowDown'}
+      iconSide="right"
+      {...buttonProps}
+    >
+      Saved queries
+    </EuiButtonEmpty>
+  );
+
+  return (
+    <EuiPopover
+      button={button}
+      isOpen={isOpen}
+      closePopover={onClose}
+      anchorPosition="downLeft"
+      panelPaddingSize="s"
+    >
+      <div style={{ width: 300, maxHeight: 400 }}>
+        <EuiText size="xs">
+          <strong>Load a saved PPL query</strong>
+        </EuiText>
+        <EuiSpacer size="s" />
+
+        {isLoading ? (
+          <div style={{ textAlign: 'center', padding: '20px' }}>
+            <EuiLoadingSpinner size="m" />
+          </div>
+        ) : savedQueries.length === 0 ? (
+          <EuiEmptyPrompt
+            iconType="search"
+            title={<h4>No saved PPL queries</h4>}
+            body={
+              <EuiText size="s" color="subdued">
+                Save a query from Explore to see it here.
```

Comment:
Not blocking:
Could be a good idea to add a link here that opens Explore in a new tab. Then add some logic to support refreshing the query list.